### PR TITLE
PriceSet - Fix metadata for extends column, for SearchKit compatibility

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes not-so-accurate metadata on the `extends` column in the PriceSet entity.

Before
----------------------------------------
Field has inconsistent format and incomplete metadata.
SearchKit says :shrug: 

After
----------------------------------------
Fixed. SearchKit says :+1: 

Technical Details
----------------------------
The `extends` field can hold one or more value, but has been following the sloppy practice of only using a serialized format if the field contains > 1 value. Otherwise it doesn't bother serializing. This PR fixes it to consistently use serialization every time, so that metadata-driven tools like SearchKit can actually understand it.

### CODE REVIEW

**Comment by seamuslee001** [comment]
Test fail seems related

File: xml/schema/Price/PriceSet.xml [hunk 1]
```diff
@@ -133,10 +133,9 @@
     <required>true</required>
     <comment>What components are using this price set?</comment>
     <pseudoconstant>
-      <table>civicrm_component</table>
-      <keyColumn>id</keyColumn>
-      <labelColumn>name</labelColumn>
+      <callback>CRM_Price_BAO_PriceSet::getExtendsOptions</callback>
     </pseudoconstant>
+    <serialize>SEPARATOR_TRIMMED</serialize>
```
Comment (eileenmcnaughton) [inline xml/schema/Price/PriceSet.xml:138]: Is this correct - my reading of the BAO is it should be `SERIALIZE_SEPARATOR_BOOKEND`

https://github.com/civicrm/civicrm-core/blob/5749f768184574a79bcbdaf3f788acb3b5d02575/CRM/Price/BAO/PriceSet.php#L48-L58

Comment (colemanw) [reply xml/schema/Price/PriceSet.xml:138]: Hmm, you're right... except that code falls over if you pass in an int, it skips the serialization process. And our sample data omits the seralization too. I'll see if I can fix it.

**Comment by eileenmcnaughton** [comment]
fail might be intermittant - test this please

**Comment by colemanw** [comment]
Retest this please

File: CRM/Core/DAO.php [hunk 2]
```diff
@@ -783,6 +783,13 @@ public function copyValues($params) {
           $this->$dbName = CRM_Core_DAO::serializeField($value, $field['serialize']);
           $allNull = FALSE;
         }
+        // When a single value was entered for a serialized field, it's probably due to sloppy coding.
+        // Folks, always use an array to pass in values for fields containing array data.
+        // Meanwhile, I'll convert it for you. You're welcome.
+        elseif (is_numeric($value) && !empty($field['serialize'])) {
```
Comment (eileenmcnaughton) [inline CRM/Core/DAO.php:789]: if we really wanted to discourage this we would add a noisy deprecation - but I suspect you are not too fussed?

File: Civi/API/Subscriber/ChainSubscriber.php [hunk 3]
```diff
@@ -90,9 +90,11 @@ protected function callNestedApi($apiKernel, &$params, &$result, $action, $entit
       $result = ['values' => [0 => $oldResult]];
     }
 
+    // Keys which should not be assumed to be chain calls
+    $blacklist = ['api.has_parent', 'api_params'];
     // Scan the params for chain calls.
     foreach ($params as $field => $newparams) {
-      if ((is_array($newparams) || $newparams === 1) && $field <> 'api.has_parent' && substr($field, 0, 3) == 'api') {
+      if ((is_array($newparams) || $newparams === 1) && !in_array($field, $blacklist) && substr($field, 0, 3) == 'api') {
```
Comment (eileenmcnaughton) [inline Civi/API/Subscriber/ChainSubscriber.php:97]: ideally use TRUE as the 3rd parameter here (strict comparison)

**Comment by eileenmcnaughton** [comment]
looks OK - I haven't done r-run test / upgrade test as yet

File: CRM/Core/DAO.php (hunk 2 above)
Comment (colemanw) [reply CRM/Core/DAO.php:789]: I was going to do that next as a followup PR. This one has already been difficult enough.

File: Civi/API/Subscriber/ChainSubscriber.php (hunk 3 above)
Comment (colemanw) [reply Civi/API/Subscriber/ChainSubscriber.php]: Ok, done.

**Comment by colemanw** [comment]
retest this please

**Pull request merged**